sources/core/Stride.Core.Design/Editor/RiderPathLocator.cs

```diff
@@ -0,0 +1,365 @@
+                                  return executablePaths;
+                          }
+                      }
+
+                      var channelFile = Path.Combine(channelDir, ".channel.settings.json");
+                      if (File.Exists(channelFile))
+                      {
+                          var json = File.ReadAllText(channelFile).Replace("active-application", "active_application");
+                          var build = ToolboxInstallData.GetLatestBuildFromJson(json);
+                          if (build != null)
+                          {
+                              var buildDir = Path.Combine(channelDir, build);
+                              var executablePaths = GetExecutablePaths(dirName, searchPattern, isMac, buildDir);
+                              if (executablePaths.Any())
+                                  return executablePaths;
+                          }
+                      }
+
+                      // changes in toolbox json files format may brake the logic above, so return all found Rider installations
+                      return Directory.GetDirectories(channelDir)
+                          .SelectMany(buildDir => GetExecutablePaths(dirName, searchPattern, isMac, buildDir));
+                  }
+                  catch (Exception e)
+                  {
+                      // do not write to Debug.Log, just log it.
+                      Logger.Warn($"Failed to get RiderPath from {channelDir}", e);
+                  }
+
+                  return new string[0];
+              })
+              .Where(c => !string.IsNullOrEmpty(c))
+              .ToArray();
+            return paths;
+        }
+
+        private static string[] GetExecutablePaths(string dirName, string searchPattern, bool isMac, string buildDir)
+        {
+            var folder = new DirectoryInfo(Path.Combine(buildDir, dirName));
+            if (!folder.Exists)
+                return new string[0];
+
+            if (!isMac)
+                return new[] { Path.Combine(folder.FullName, searchPattern) }.Where(File.Exists).ToArray();
+            return folder.GetDirectories(searchPattern).Select(f => f.FullName)
+              .Where(Directory.Exists).ToArray();
+        }
+
+        // Disable the "field is never assigned" compiler warning. We never assign it, but Unity does.
+        // Note that Unity disable this warning in the generated C# projects
+#pragma warning disable 0649
+
+        [Serializable]
+        class SettingsJson
+        {
+            public string install_location;
+      
+            //[CanBeNull]
+            public static string GetInstallLocationFromJson(string json)
+            {
+                try
+                {
+                    return JsonConvert.DeserializeObject<SettingsJson>(json).install_location;
+                }
+                catch (Exception)
+                {
+                    Logger.Warn($"Failed to get install_location from json {json}");
```

Comment:
This method has an optional exception parameter, might as well pass an instance of the exception to it.